studio/components/interfaces/Database/Migrations/Migrations.tsx

```diff
@@ -0,0 +1,162 @@
+
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import Table from 'components/to-be-cleaned/Table'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import { DatabaseMigration, useMigrationsQuery } from 'data/database/migrations-query'
+import { Editor } from '@monaco-editor/react'
+import CodeEditor from 'components/ui/CodeEditor'
+
+const Wrappers = () => {
+  const [search, setSearch] = useState('')
+  const [selectedMigration, setSelectedMigration] = useState<DatabaseMigration>()
+
+  const { project } = useProjectContext()
+  const { data, isLoading, isSuccess, isError, error } = useMigrationsQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+  const migrations =
+    search.length === 0
+      ? data?.result ?? []
+      : data?.result.filter((migration) => migration.version.includes(search)) ?? []
+
+  return (
+    <>
+      <div>
+        <div className="flex items-center justify-between mb-6">
+          <div>
+            <h3 className="mb-2 text-xl text-scale-1200">Migrations</h3>
+            <div className="text-sm text-scale-900">
+              History of migrations that have been run on your database
+            </div>
+          </div>
+          <div className="flex items-center space-x-2">
+            <Link href="https://supabase.com/docs/guides/getting-started/local-development#database-migrations">
+              <a target="_blank" rel="noreferrer">
+                <Button type="default" icon={<IconExternalLink strokeWidth={1.5} />}>
+                  Documentation
+                </Button>
+              </a>
+            </Link>
+          </div>
+        </div>
+
+        {isLoading ? (
+          <div className="space-y-2">
+            <ShimmeringLoader />
+            <ShimmeringLoader className="w-3/4" />
+            <ShimmeringLoader className="w-1/2" />
+          </div>
+        ) : (
+          <div>
+            {isError && (
+              <Alert
+                withIcon
+                variant="warning"
+                title="Failed to retrieve migration history for database"
+                actions={[
+                  <Link
+                    key="contact-support"
+                    href={`/support/new?ref=${project?.ref}&category=dashboard_bug&subject=Unable%20to%20view%20database%20migrations`}
+                  >
+                    <a>
+                      <Button type="default" className="ml-4">
+                        Contact support
+                      </Button>
+                    </a>
+                  </Link>,
+                ]}
+              >
+                <p className="mb-1">
+                  Try refreshing your browser, but if the issue persists, please reach out to us via
+                  support.
+                </p>
+                <p>Error: {(error as any)?.message ?? 'Unknown'}</p>
```

Comment:
Is it worth checking if the error here is the table not existing and then pointing the user towards the docs on how to run migrations?

Reply:
oh yeah good point - lemme think on how to handle this case